judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/hibernate/ContestHibernateDao.java

```diff
@@ -49,10 +50,26 @@ public ContestHibernateDao(SessionFactory sessionFactory, Clock clock, ActorProv
     }
 
     @Override
-    public Page<ContestModel> selectPagedByUserJid(String userJid, SelectionOptions options) {
-        return selectPaged(new FilterOptions.Builder<ContestModel>()
-                .addCustomPredicates(hasViewerOrAbove(userJid))
-                .build(), options);
+    public Page<ContestModel> selectPaged(SearchOptions searchOptions, SelectionOptions options) {
+        FilterOptions.Builder<ContestModel> filterOptions = new FilterOptions.Builder<ContestModel>();
```

Comment:
It's much cleaner if we instead have `createFilterOptions()` instead. So the code will be something like

```
return selectPaged(createFilterOptions(searchOptions).build(), options);
```

(similarly for the next method, we call `addCustomPredicate()` before `build()`)